dwl.c

```diff
@@ -1676,23 +1673,17 @@ rendermon(struct wl_listener *listener, void *data)
 
 	/* Render if no XDG clients have an outstanding resize and are visible on
 	 * this monitor. */
-	/* Checking m->un_map for every client is not optimal but works */
-	wl_list_for_each(c, &clients, link) {
-		if ((c->resize && m->un_map) || (c->type == XDGShell
-				&& (c->surface.xdg->pending.geometry.width !=
-				c->surface.xdg->current.geometry.width
-				|| c->surface.xdg->pending.geometry.height !=
-				c->surface.xdg->current.geometry.height))) {
-			/* Lie */
-			wlr_surface_send_frame_done(client_surface(c), &now);
+	wl_list_for_each(c, &clients, link)
+		if (!c->isfloating && c->resize) {
```

Comment:
Instead checking `c->isfloating` I'd check `VISIBLEON(c, m)`

Reply:
That doesn't work because then resizing willl go back to flickering and 2 VISIBLEON doesn't actually check if a client is visible on a monitor. It only checks c->mon == m and if the tags are correct, which means that it will ignore unresized clients on another monitor. This is why the 'overshoot' happens. If you look at the originial implementation, there is also overshoot. This is related to #22

Reply:
I don't understand why we have to skip frames in this monitor if a client is resizing on another monitor, but omitting that, you said `we only care about 'loose' resizes when something is resizing and by extension lfoating` and here you are checking if the client is NOT floating

Reply:
Because a client doesn't have to follow monitor bounds. This is what I mean when I'm talking about 'overshoot'. Basically the client doesn't resize fast enough and appears on the next monitor. This is to prevent that. If VISIBLEON actually checked visibilty, I would keep it. But VISIBLEON only checks formal assignment. So c->mon == m and the tags.

Yes, I'm checking if the client is not floating. If the client is floating, we let it do whatever it wants, and only when the client is actually important, as in stacking, we want to check if it's correct.

Reply:
```c
int on_mon = 0;
struct wlr_surface_output *s = NULL;
if (!c->scene->state.enabled)
	continue;
wl_list_for_each(s, &client_surface(c)->current_outputs, link)
	if ((on_mon = s->output == m->wlr_output))
		break;
```

Reply:
Thanks for that solution. Although if wlroots already checks that, maybe closing #22 would be a good idea?

Reply:
I'm sorry, nvm, I just saw that #22 is already closed

Reply:
@Sevz17 @djpohly What do you think about making this a macro? PIXELONMON or something like that?